great_expectations/data_context/store/profiler_store.py

```diff
@@ -0,0 +1,63 @@
+import random
+import uuid
+from typing import Union
+
+from great_expectations.data_context.store.configuration_store import ConfigurationStore
+from great_expectations.data_context.types.resource_identifiers import (
+    ConfigurationIdentifier,
+    GeCloudIdentifier,
+)
+from great_expectations.rule_based_profiler.config import RuleBasedProfilerConfig
+
+
+class ProfilerStore(ConfigurationStore):
+    """
+    A ProfilerStore manages Profilers for the DataContext.
+    """
+
+    _configuration_class = RuleBasedProfilerConfig
+
+    def serialization_self_check(self, pretty_print: bool) -> None:
+        """
+        Fufills the abstract method defined by the parent class.
+        See `ConfigurationStore` for more details.
+        """
```

Comment:
Mirror the logic in CheckpointStore. I'd like to move all this "pretty-printing" to logging if possible (don't love the `if pretty_print` checks)

Reply:
It looks like all this logic could be in the base class, especially if the ProfilerStore called this method with an instance of RuleBasedProfilerConfig (or provided an instance by overriding a method in the base class)?

Reply:
I agree. For purposes of unblocking the team, could I put that refactor in a separate ticket?

Reply:
okay

Reply:
Wrote ticket; resolving convo